docs/src/main/docs/guides/05_Dockerfile.adoc

```diff
@@ -281,13 +287,143 @@ modules to the `jlink` command:
      "--add-modules", \
      "java.base,java.logging,java.sql,java.desktop,java.management,java.naming,jdk.unsupported", \
      "--output", "/var/tmp/myjre"]
 ----
 
-=== How Big is This Stuff?
+
+== Creating a Docker Image Using Jib
+
+https://github.com/GoogleContainerTools/jib[Jib] is a collection of build tools for
+constructing Docker images for Java applications. Jib allows you to build images
+straight from a Maven (or Gradle) plugin without the need for the `docker`
+command. It also uses a
+https://github.com/GoogleContainerTools/distroless[distroless] base image that
+can result in a smaller final image.
+
+We'll be using the `jib-maven-plugin` in this example. With
+`jib-maven-plugin` you don't need a Dockerfile
+and it creates a final Docker image that follows a set
+of best-practices for packaging Java applications in a container.
+
+Docker is only required if you are going to install the image locally.
+It is *not* required to build and _push the image_ to a Docker Registry,
+ such as Docker Hub. This can be accomplished via `mvn jib:build` after
+https://github.com/GoogleContainerTools/jib/tree/master/jib-maven-plugin#configuration[configuring which
+Registry to connect to].
+This fits well in an environment where administrative permissions
+ are unavailable on developer machines to install or run Docker.
+
+Add the following plugin to the `pom.xml`. This adds a goal to
+the Maven `package` phase that containerizes the application
+using a locally available Docker daemon.
+
+[source,xml,subs="verbatim,attributes"]
+.pom.xml
+----
+<project>
+  ...
+  <properties>
+    <jib-maven-plugin.version>0.10.1</jib-maven-plugin.version>
+  </properties>
+  ...
+  <build>
+    <plugins>
+      ...
+      <plugin>
+          <groupId>com.google.cloud.tools</groupId>
+          <artifactId>jib-maven-plugin</artifactId>
+          <version>${jib-maven-plugin.version}</version>
+          <configuration>
+              <to>
+                  <image>jib-${project.artifactId}</image>
+                  <tags>
+                      <tag>${project.version}</tag>
+                      <tag>latest</tag>
+                  </tags>
+              </to>
+              <container>
+                  <!-- good defaults intended for containers -->
+                  <jvmFlags>
+                      <jmxFlag>-server</jmxFlag>
+                      <jmxFlag>-Djava.awt.headless=true</jmxFlag>
+                      <jmxFlag>-XX:+UnlockExperimentalVMOptions</jmxFlag>
+                      <jmxFlag>-XX:+UseCGroupMemoryLimitForHeap</jmxFlag>
+                      <jmxFlag>-XX:InitialRAMFraction=2</jmxFlag>
+                      <jmxFlag>-XX:MinRAMFraction=2</jmxFlag>
+                      <jmxFlag>-XX:MaxRAMFraction=2</jmxFlag>
+                      <jmxFlag>-XX:+UseG1GC</jmxFlag>
+                      <jmxFlag>-XX:MaxGCPauseMillis=100</jmxFlag>
+                      <jmxFlag>-XX:+UseStringDeduplication</jmxFlag>
+                  </jvmFlags>
```

Comment:
These JVM flags could probably be removed, and left as an excersize for the reader.

Reply:
Done.